### DISCUSSION TITLE
Shape Inference with Custom Operators

### DISCUSSION THREAD

**Discussion by erickksun** [microsoft/onnxruntime#22966] [API Q&A]
hi,
I have an ONNX model that includes a custom operator, which I implemented using ONNX Runtime Extension. The implementation functions as expected. However, I am facing an issue with performing shape inference and saving the results back into the ONNX file. I’ve attempted to use the tool found at [this link](https://github.com/microsoft/onnxruntime/blob/main/onnxruntime/python/tools/symbolic_shape_infer.py), but it appears not to support custom operators.

**Comment by skottmckay** [comment]
Why do you need to save the results back to the ONNX file? Shape inferencing runs during model load, so if the shapes can be inferred there's generally no need to save them previously.
